### DISCUSSION TITLE
OAuth initiated from Seller Central

### DISCUSSION THREAD

**Discussion by TimOBannon** [amzn/selling-partner-api-models#3205] [General]
Looking at this part of the docs - https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#step-2-the-selling-partner-consents-to-authorize-your-application

If the seller starts the OAuth flow from within Seller Central, Amazon will redirect to my Login URI after the seller authorizes my application.  When I get the redirect to my Login URI, how do I know which region is being authorized?  Would I have to derive this from the referer?  The docs only mention the following query params:
- amazon_callback_uri
- amazon_state
- selling_partner_id

**Comment by teddy-codes** [comment]
I would probably just use this: https://github.com/amzn/selling-partner-api-docs/blob/main/references/sellers-api/sellers.md#getmarketplaceparticipations

**Reply by TimOBannon** [reply]
Thanks for the reply, but I don't know that would help here.  Since I'm in the middle of the OAuth flow, I haven't been authorized to request information about the seller yet.  Even if I was, I don't need the list of all the seller's marketplaces - I need the marketplace the seller logged into, since authorization happens at the region level.

For example, it would not help me to know that the seller operates in the US, UK, and JP - each requires its own refresh token, and I would not know which consent was given.

**Reply by mbtoldya** [reply]
Tim - I am stuck in the same spot as you.  It seems with this approach we are building a workflow that is intended to start at the partner appstore instead our OWN site.  Therefore, we are mixed up and need to use https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow - the problem being that they do NOT send the data we are after.  Most likely this has to do with the state of our app.
